retroshare-gui/src/gui/chat/ChatWidget.cpp

```diff
@@ -393,30 +430,99 @@ void ChatWidget::init(const ChatId &chat_id, const QString &title)
 				if (time(nullptr) <= historyIt->recvTime+2)
 					continue;
 
-				QString name;
-				if (chatId.isLobbyId() || chatId.isDistantChatId())
-				{
-					RsIdentityDetails details;
-					time_t start = time(nullptr);
-					while (!rsIdentity->getIdDetails(RsGxsId(historyIt->peerName), details))
-					{
-						std::this_thread::sleep_for(std::chrono::milliseconds(10));
-						if (time(nullptr)>start+2)
-						{
-							std::cerr << "ChatWidget History haven't found Id Details and have wait 1 sec for it." << std::endl;
-							break;
+				const std::string
+					notbeinghashableisagoodidea((const char*)historyIt->chatPeerId.toByteArray());
+				struct lookup_info& state = statecache[notbeinghashableisagoodidea];
+				if(!state.hasName) {
+					int tries;
+					for(tries=0;tries<4;++tries) {
+						assert(!state.hasName);
+						if(state.isDirect) {
+						GOT_DIRECT:
```

Comment:
these labels are a bit misleading: we're not in a switch and there's no goto going there. If you want to mark some par of the code with some kind of milestone or so, keep this but comment out the line so we know it's not actually used.

Reply:
ah forget this. I didn't see the goto later on.